php/api/endpoint/class-audit.php

```diff
@@ -245,9 +245,23 @@ public function rest_results_get( $rest_post, $field_name, $request ) {
 		$results = array();
 
 		foreach ( $standards as $standard ) {
-			$meta                 = get_post_meta( $rest_post['id'], sprintf( '_audit_%s', $standard ), true );
+			$meta = get_post_meta( $rest_post['id'], sprintf( '_audit_%s', $standard ), true );
+
+			// If we don't have any data for the standard then there's no point in adding an empty element to the results.
+			if ( empty( $meta ) ) {
+				continue;
+			}
+
 			$results[ $standard ] = $meta;
 
+			/*
```

Comment:
@danlouw We're starting over and forcing all themes/plugins to be re-audited, so this code doesn't seem necessary to me. I don't think we should support backwards compatibility for scoring.

Reply:
Cool, thanks @valendesigns, makes sense. Just to clarify:

As it currently stands:
* `rest_results_get()` will return whatever results data it finds in meta
* `rest_results_update()` will store whatever results data is passed to it in meta (provided the standard it is passed under is allowed).

So if scores are passed, it will be saved in meta, if it isn't it won't.
Similarly if scores are found in meta, it will be returned, if it isn't it won't.

If we are not doing anything to explicitly prevent scores from entering meta or being returned in result, then:

* We are assuming it’s up to the Audit Server to get rid of scores. If no score comes from AS, API will have no score to save, but if AS does supply scores again for any reason, API will start saving and returning it again
* We are also assuming all historical data will be blown away. Even if AS stops providing scores now, any score data that remains in meta from previous audits will still be returned by the API

Since the JIRA description says “Remove scoring and rating meta from API plugin”, I thought we wanted to follow a more proactive approach to get rid of existing scores in meta.

But if this is not the case, then as far as I can see there’s really nothing we need to do regarding scores in the API - we’re just relying on the Audit Server to not pass the API any scores.

So next steps I’ll focus on then are:
* Revert code that strips scores from results in API
* Remove ratings field from CPT (I assume this is still relevant)
* Revert code that removes empty standard properties from results and rather address that separately in TIDE-432, since I see we’ll need to also update unit tests to accommodate this

Reply:
Sounds good, thanks for looking into it. I think moving forward we will do a backup of the DB and dump the posts and meta tables to start over fresh. Looking forward to the updated PR.